### ISSUE TITLE
Integrate Essentia/MusiCNN for music embeddings

### ISSUE THREAD

**Issue by james-westwood** [james-westwood/playchitect#5]
## Overview
Integrate Essentia with pre-trained MusiCNN model to extract 128-dimension embeddings for richer clustering.

## Tasks
- [ ] Install Essentia with TensorFlow support
- [ ] Download pre-trained MusiCNN model from Essentia model zoo
- [ ] Create `playchitect/core/embedding_extractor.py` module
- [ ] Implement embedding extraction (128-dim vectors)
- [ ] Add caching system for embeddings
- [ ] Concatenate embeddings with intensity features
- [ ] Update clustering to handle 136-dim vectors (8 + 128)
- [ ] Add genre auto-detection using pre-trained classifier
- [ ] Create unit tests

## Technical Details

**Model**: MusiCNN from Essentia model zoo
- **Size**: ~790k parameters
- **Output**: 128-dimension embedding vector
- **Pre-trained on**: Million Song Dataset (50 most common tags)

**Feature Vector** (136 dimensions):
```
[8 intensity features + 128 embedding features]
```

**Installation**:
```bash
pip install essentia-tensorflow
```

**Usage**:
```python
from essentia.standard import TensorflowPredictMusiCNN
model = TensorflowPredictMusiCNN(graphFilename="model.pb")
embeddings = model(audio)
```

## Acceptance Criteria
- [ ] Essentia integrated and working
- [ ] 128-dim embeddings extracted successfully
- [ ] Clustering works with 136-dim features
- [ ] Genre auto-detection functional
- [ ] Caching speeds up re-analysis
- [ ] Unit tests with mock models

## Resources
- [Essentia models](https://essentia.upf.edu/models.html)
- [MusiCNN paper](https://www.researchgate.net/publication/341084851_Tensorflow_Audio_Models_in_Essentia)
- [Essentia TensorFlow tutorial](https://essentia.upf.edu/tutorial_tensorflow_auto-tagging_classification_embeddings.html)

## Depends On
- #2 (Clustering Integration)

## Blocks
- #6 (Genre-aware clustering)
